src/python/pants/backend/terraform/target_types.py

```diff
@@ -51,5 +57,71 @@ class TerraformModuleTarget(Target):
     )
 
 
+class TerraformBackendConfigField(OptionalSingleSourceField):
+    alias = "backend_config"
+    help = "Configuration to be merged with what is in the configuration file's 'backend' block"
+
+    def empty(self) -> TerraformBackendConfigField:
+        """Clear the backend config to ensure that."""
+        return TerraformBackendConfigField(None, self.address)
+
+
+class TerraformVarFilesField(MultipleSourcesField):
+    alias = "var_files"
+    expected_file_extensions = (".tfvars",)
+    help = generate_multiple_sources_field_help_message(
+        "Example: `var_files=['common.tfvars', 'prod.tfvars']`"
+    )
+
+
+class TerraformExtraArgs(StringSequenceField):
```

Comment:
Are these really extra command line arguments or some variable overrides?

I would suggest to follow a similar approach as the one in the Helm backend, where the `HelmDeploymentTarget` allows fine tuning of the different deployment settings via individual fields (instead of a generic field for passthrough arguments) but still allowing the usage of passthrough args via the `HelmSubsystem`.

You can see there that the `HelmDeploymentTarget` also allows for inline value overrides (which maybe is what field is trying to achieve?) supporting even value interpolation of environment variables.

Reply:
You're right, I think most of the args here would be better as passthrough args for the tool (eg "-auto-approve", "-parallelism"). The only one which is more tied to an individual deployment would be "-var". I could make that into a field, like `HelmDeploymentValuesField`.

Reply:
I find having those in a field much more user friendly, so it'd be a great improvement.

I wouldn't bother with value interpolation at this point though, that always can be added later.